_data-prepper/common-use-cases/log-enrichment.md

```diff
@@ -0,0 +1,398 @@
+
+The following example combines the `split_string` and `key_value` processors to extract query parameters from an Apache log line:
+
+```json
+pipeline
+ ...
+  processor:
+    - grok:
+      match:
+        message: [ "%{COMMONAPACHELOG_DATATYPED}" ]
+    - split_string:
+      entries:
+        - source: request
+          delimiter: "?"
+    - key_value:
+      source: "/request/1"
+      field_split_characters: "&"
+      value_split_characters: "="
+      destination: query_params
+```
+{% include copy-curl.html %}
+
+## Mutating events
+
+The different [Mutate event]({{site.url}}{{site.baseurl}}/data-prepper/pipelines/configuration/processors/mutate-event/) processors let you rename, copy, add, and delete event entries.
+
+In this example, the first processor sets the value of the `debug` key to `true` if the key already exists in the event. The second processor only sets the `debug` key to `true` if the key doesn't exist in the event, because `overwrite_if_key_exists` is set to `true`.
+
+```json
+...
+processor:
+  - add_entries:
+    entries:
+    - key: "debug"
+      value: true 
+...
+processor:
+  - add_entries:
+    entries:
+    - key: "debug"
+      value: true 
+      overwrite_if_key_exists: true
+...
+```
+{% include copy-curl.html %}
+
+You can also use a format string to construct new entries from existing entries. For example, `${date}-${time}` will create a new entry based on the values of the existing entries `date` and `time`.
+
+For example, the following pipeline adds new event entries dynamically from existing events:
+
+```json
+processor:
+  - add_entries:
+    entries:
+    - key: "key_three"
+      format: "${key_one}-${key_two}
+```
+{% include copy-curl.html %}
+
+For example, consider the following incoming event:
+
+```json
+{
+   "key_one": "value_one",
+   "key_two": "value_two"
+}
+```
+{% include copy-curl.html %}
+
+The processor transforms it into an event with a new key `key_three`, which combines values of other keys in the original event, as shown in the following example:
+
+```json
+{
+   "key_one": "value_one",
+   "key_two": "value_two",
+   "key_three": "value_one-value_two"
+}
+```
+{% include copy-curl.html %}
+
+## Mutating strings
+
+The various [Mutate string]({{site.url}}{{site.baseurl}}/data-prepper/pipelines/configuration/processors/mutate-string/) processors offer tools to manipulate strings in incoming data. For example, if you need to split a string into an array, use the `split_string` processor:
+
+```json
+...
+processor:
+  - split_string:
+    entries:
+    - source: "message"
+      delimiter: "&"
+...
+```
+{% include copy-curl.html %}
+
+The processor will transform a string such as `a&b&c` into `["a", "b", "c"]`.
+
+## Converting lists to maps
+
+The [List-to-map]({{site.url}}{{site.baseurl}}/data-prepper/pipelines/configuration/processors/list-to-map/) processor, which is one of the Mutate events processors, converts a list of objects in an event to a map.
```

Comment:
Can "Mutate event" be formatted so as to avoid any mid-sentence capitalization confusion? Last phrase: "converts a list of event objects to a map"?

Reply:
In researching the naming convention "mutate event processor," I didn't come across it as a proper name. Instead, it appears as a common name for processors that convert an event.